### PR TITLE
ClarifyPureAsDiscussed

### LINKED ISSUE #2971: pure(...) operator semantic clarification
Besides the grammar, the pure operator is only mentioned in chapter 12.3

> pure(impureFunctionCall(. . .)) – which allows calling impure functions in any pure context.

The operator could use some clarification. Should it not be expanded on in chapter 3.7 along with the other operators?

Some concrete questions:
1. Are there any semantic limitations? Is any expression legal as argument?
2. Does it apply to all functions in the argument? For example, is `pure(f(g()))` equivalent to `pure(f(pure(g())))`?
3. Does it mean we can rearrange the evaluation order in sub-expressions? For example, in `pure(f()+g())` can `g()` be evaluated before `f()`?

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #2971

### CODE REVIEW

File: chapters/functions.tex [hunk 1]
```diff
@@ -316,6 +316,7 @@ \section{Pure Modelica Functions}\label{pure-modelica-functions}
   A \lstinline!when!-statement.
 \item
   \lstinline!pure(impureFunctionCall($\ldots$))! -- which allows calling impure functions in any pure context.
```
Comment (henrikt-ma) [inline chapters/functions.tex:318]: The name of the callee shouldn't be `impureFunctionCall`:
```suggestion
  \lstinline!pure(impureFunction($\ldots$))! -- which allows calling impure functions in any pure context.
```

File: chapters/functions.tex [hunk 2]
```diff
@@ -316,6 +316,7 @@ \section{Pure Modelica Functions}\label{pure-modelica-functions}
...
+  Note that the argument to \lstinline!pure! must be a function call, it only applies to the call and not to the arguments of the function call.
```
Comment (henrikt-ma) [inline chapters/functions.tex:319]: This is more than just a note:
```suggestion
  The wrapping in \lstinline!pure($\ldots$)! only by-passes the purity checking of the callee \lstinline!impureFunction!; the argument expressions of the function call are not affected.
```

**Review by henrikt-ma — CHANGES_REQUESTED** [review]
With a more direct formulation this looks OK to me, but lacking a really clear conclusion in #2971 it would be good to also get an approval from @jonathankampemodelon.

Edit: At least some kind of positive feed-back from @jonathankampemodelon, as he can't be added as reviewer.

**Review by henrikt-ma — COMMENTED** [review]
I am OK with the way this is written, but it would be good to "also" get an approval from someone else before merging.

**Review by henrikt-ma — APPROVED** [review]

**Review by sjoelund — APPROVED** [review]
Seems OK. I tried it in OpenModelica, but apparently we don't even parse `pure` as a function call so this is not used by any library I know of and would not cause problems either way it is clarified...

**Pull request merged**